### PR TITLE
Time io update

### PR DESCRIPTION
**Pull request by uramirez8707**
**Description**
This PR:
 1.  Update to write out the time data based on how FMS was compiled to reproduce previous diag manager behavior
 2.  Implement capability to specify which time in a time average period to use when setting filename and adds tests for it

Fixes # (issue)

**How Has This Been Tested?**
CI, including updated tests

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Review by thomas-robinson — CHANGES_REQUESTED** [review]
At a minimum, you need to add 4 spaces to the end of the "double    " string when you set it.  I think you should also add the namelist flag for reproducing the old behavior.

File: diag_manager/diag_data.F90 [hunk 1]
```diff
@@ -392,7 +395,12 @@ MODULE diag_data_mod
   REAL :: FILL_VALUE = 9.9692099683868690e+36
 #endif
 
+  !! @note `pack_size` and `pack_size_str` are set in diag_manager_init depending on how FMS was compiled
+  !! if FMS was compiled with default reals as 64bit, it will be set to 1 and "double",
+  !! if FMS was compiled with default reals as 32bit, it will set to 2 and "float"
+  !! This is to reproduce previous diag manager behavior.
   INTEGER :: pack_size = 1 !< 1 for double and 2 for float
+  CHARACTER(len=10) :: pack_size_str="double" !< Pack size as a string to be used in fms2_io register call
```
[the remark this answers is not in the document]
Comment (uramirez8707) [reply diag_manager/diag_data.F90:403]: why?

File: diag_manager/diag_data.F90 [hunk 2]
```diff
@@ -392,7 +395,12 @@ MODULE diag_data_mod
   REAL :: FILL_VALUE = 9.9692099683868690e+36
 #endif
 
+  !! @note `pack_size` and `pack_size_str` are set in diag_manager_init depending on how FMS was compiled
+  !! if FMS was compiled with default reals as 64bit, it will be set to 1 and "double",
+  !! if FMS was compiled with default reals as 32bit, it will set to 2 and "float"
+  !! This is to reproduce previous diag manager behavior.
```
[the remark this answers is not in the document]
Comment (uramirez8707) [reply diag_manager/diag_data.F90:401]: what is the new behavior?

File: diag_manager/fms_diag_yaml.F90 [hunk 3]
```diff
@@ -513,6 +518,10 @@ subroutine fill_in_diag_files(diag_yaml_id, diag_file_id, fileobj)
          is_optional=.true.)
   call set_new_file_freq(fileobj, freq_buffer, buffer)
 
+  deallocate(buffer)
```
[the remark this answers is not in the document]
Comment (uramirez8707) [reply diag_manager/fms_diag_yaml.F90:521]: The buffer always gets allocated in `diag_get_value_from_key`, there is no need to check

File: diag_manager/diag_data.F90 (hunk 1 above)
[the remark this answers is not in the document]
Comment (thomas-robinson) [reply diag_manager/diag_data.F90:403]: Some compilers require this when setting a variable.

File: diag_manager/diag_data.F90 (hunk 2 above)
[the remark this answers is not in the document]
Comment (thomas-robinson) [reply diag_manager/diag_data.F90:401]: we should discuss this more

File: diag_manager/diag_data.F90 (hunk 1 above)
[the remark this answers is not in the document]
Comment (uramirez8707) [reply diag_manager/diag_data.F90]: https://github.com/NOAA-GFDL/FMS/pull/1129/commits/3ac08c6eca17f8d1c85c55818d7237983fd22427

**Pull request merged**